.github/workflows/ci.yml

```diff
@@ -212,15 +212,23 @@ jobs:
       - name: "Install Rust toolchain"
         run: rustup show
 
-      - uses: astral-sh/setup-uv@6b9c6063abd6010835644d4c2e1bef4cf5cd0fca # v6.0.1
-      - name: "Install required Python versions"
-        run: uv python install
-
       - name: "Install cargo nextest"
         uses: taiki-e/install-action@ab3728c7ba6948b9b429627f4d55a68842b27f18 # v2.50.3
         with:
           tool: cargo-nextest
 
+      - name: "Build the uv binary"
```

Comment:
We were using an earlier version of uv to install Python. But this breaks tests with the new install/upgrade mechanism. So we're building the binary with the latest version and using that instead.

Reply:
I would rather not do this if we can avoid it. How does this break the tests? The tests should be isolated from how Python is installed.

Reply:
> This affects tests that depend on our managed python-build-standalone installations. For example, virtual environments now point to the symlink directory/junction, which wouldn't exist with mainline uv installations.

What's the exact failure mode? Virtual environments point to an invalid directory? Or snapshots change? Does this mean that this change is breaking for any users on existing managed Python versions?

Reply:
I've been investigating this further and it looks like these failures were useful. I'm going to revert the `ci.yml` change and work on a fix. 

I marked the PR as `breaking` but it was really just supposed to be that virtual environments created with an older version of uv wouldn't transparently upgrade until recreated.